ingestion_server/ingestion_server/elasticsearch_models.py

```diff
@@ -320,20 +320,23 @@ def database_row_to_elasticsearch_doc(row, schema):
         attrs = Audio.get_instance_attrs(row, schema)
         popularity = attrs["standardized_popularity"]
 
+        length = Audio.get_length(row[schema["duration"]])
+
         return Audio(
             bit_rate=row[schema["bit_rate"]],
             sample_rate=row[schema["sample_rate"]],
             genres=row[schema["genres"]],
             category=row[schema["category"]],
             duration=row[schema["duration"]],
+            length=length,
             authority_boost=authority_boost,
             max_boost=max(popularity or 1, authority_boost or 1),
             min_boost=min(popularity or 1, authority_boost or 1),
             **attrs,
         )
 
     @staticmethod
-    def get_duration(duration):
+    def get_length(duration):
         if not duration:
             return None
         for length in Audio.Durations:
```

Comment:
I have to be honest, I don't quite understand this function 😅 wouldn't this `duration < length` check include files at a certain size category _and all files below that size category_ in a search? It seems that if I searched for "long" files, I would get all "medium" and below too since those are `< "long"`.

Reply:
I had that same though too, but it relies on the order of the audio durations enum and also the fact that if something is labelled "medium" it won't get included in the query for specifically ones labelled "long" because it's a plain keyword search (the search doesn't know the ordinal relationship between the terms).

The ordering matters because it iterates and finds the first one that the duration is _shorter_ than. If it went longest to shortest enum then everything would be long, of course. Otherwise they get checked in order and return the ones that they're shorter than.

However, there's a bug in this PR that will prevent the shortest time from working, it isn't present in the `Audio.Durations` enum.

Reply:
Sorry for not explaining earlier! I have fixed the bug by adding the missing range.